File: C/Exercice2_1.c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

int main()
{
    __pid_t pid_fils;
    pid_fils = fork();
    if (pid_fils == -1)
    {
        puts("Erreur lors de la creation du nouveau processus\n");
        exit(1);
    }
    if (pid_fils == 0)
    {
        printf("Je suis le fils et mon pid est: %d\n", getpid());
    }
    else
    {
        printf("Je suis le pere et mon pid est: %d\n", getpid());
    }
    
    return 0;
}

File: Python/Exercice2_1.py
#coding: utf-8
import os
import sys

pid_fils = os.fork()

if pid_fils == -1:
    print("Erreur lors de la creation du nouveau processus\n")
    sys.exit(1)
    
if pid_fils == 0:
    print(f"Je suis le fils et mon pid est: {os.getpid()}")

else:
    print(f"Je suis le pere et mon pid est: {os.getpid()}")
